ttnn/tensor.py

```diff
@@ -245,6 +297,60 @@ def impl(ttl_tensor, layout):
     return Tensor(ttl_tensor)
 
 
+# TODO:integrate the move op to convert DRAM to L1 and L1 to DRAM
+@decorate_operation()
+def to_memory_config(tensor, memory_config: MemoryConfig):
+    """
+    to_mem_config(tensor: ttnn.Tensor) -> Tensor
+
+    Converts a tensor to the desired mem_config, used for converting tensors to sharded tensors or interleaved
+
+
+    Args:
+        * :attr:`tensor`: the ttnn.Tensor
+
+    Example::
+        >>> device_id = 0
+        >>> device = ttnn.open(device_id)
+        >>> tensor = ttnn.to_device(ttnn.from_torch(torch.randn((10, 64, 32), dtype=torch.bfloat16)), device)
+        >>> tensor = ttnn.to_mem_config(tensor, memory_config)
+    """
+
+    ttl_tensor = tensor.value
+    # to_sharded path
+    if memory_config.is_sharded():
+        if ttl_tensor.is_sharded():
```

Comment:
Can the mem config be changing the shard type? like input is height sharded, and want to change to block sharded, etc.?

Reply:
I can add that capability, if we are supplying a memconfig of a sharded tensor, and the tensor is already sharded.
Underneath the hood we can do sharded_to_interleaved and interleaved_to_sharded